GVRf/Framework/src/org/gearvrf/scene_objects/GVRCylinderSceneObject.java

```diff
@@ -215,104 +215,173 @@ private void createCap(float radius, float height, int numSlices,
         }
     }
 
-    private void createBody(float bottomRadius, float topRadius, float height, int numStacks, int numSlices) {
+    private void createBody(float bottomRadius, float topRadius, float height,
+            int stackNumber, int sliceNumber) {
         float difference = bottomRadius - topRadius;
         float halfHeight = height / 2.0f;
 
-        for(int stack = 0; stack < numStacks; stack++) {
-            float stackPercentage0 = ((float)(stack)/numStacks);
-            float stackPercentage1 = ((float)(stack+1)/numStacks);
+        for (int stack = 0; stack < stackNumber; stack++) {
+
+            int initVertexCount = vertexCount;
+
+            float stackPercentage0 = ((float) (stack) / stackNumber);
+            float stackPercentage1 = ((float) (stack + 1) / stackNumber);
 
             float t0 = 1.0f - stackPercentage0;
             float t1 = 1.0f - stackPercentage1;
             float y0 = -halfHeight + (stackPercentage0 * height);
             float y1 = -halfHeight + (stackPercentage1 * height);
 
-            for(int slice = 0; slice < numSlices; slice++) {
-                float slicePercentage0 = ((float)(slice)/numSlices);
-                float slicePercentage1 = ((float)(slice+1)/numSlices);
-                float theta0 = slicePercentage0*2.0f*(float)Math.PI;
-                float theta1 = slicePercentage1*2.0f*(float)Math.PI;
-                float cosTheta0 = (float)Math.cos(theta0);
-                float sinTheta0 = (float)Math.sin(theta0);
-                float cosTheta1 = (float)Math.cos(theta1);
-                float sinTheta1 = (float)Math.sin(theta1);
+            float nx, ny, nz;
+            for (int slice = 0; slice < sliceNumber; slice++) {
+                float slicePercentage0 = ((float) (slice) / sliceNumber);
+                float slicePercentage1 = ((float) (slice + 1) / sliceNumber);
+                double theta0 = slicePercentage0 * 2.0 * Math.PI;
+                double theta1 = slicePercentage1 * 2.0 * Math.PI;
+                double cosTheta0 = Math.cos(theta0);
+                double sinTheta0 = Math.sin(theta0);
+                double cosTheta1 = Math.cos(theta1);
+                double sinTheta1 = Math.sin(theta1);
 
                 float radius = (bottomRadius - (difference * stackPercentage0));
```

Comment:
This is a quibble, but: Now that the trig values are double, you are doing a float->double conversion every time you multiply a float by a double. That is, it would be a bit faster if `radius` were also a `double` ... probably no real impact on precision, though.